src/__tests__/unit/decorators/grpc.decorator.unit.ts

```diff
@@ -5,39 +5,97 @@
 
 import {MetadataInspector} from '@loopback/metadata';
 import {expect} from '@loopback/testlab';
+import {lastValueFrom, Observable, PartialObserver} from 'rxjs';
 import {grpc, GRPC_METHODS} from '../../..';
 import {
-  Greeter,
-  HelloReply,
-  HelloRequest,
   TestRequest,
-  TestReply,
-} from '../../acceptance/greeter.proto';
+  TestResponse,
+  protoMetadata,
+  Greeter,
+} from '../../fixtures/server/greeter';
 
 describe('@rpc decorator', () => {
   it('defines reflection metadata for rpc method', () => {
-    class GreeterCtrl implements Greeter.Service {
-      @grpc(Greeter.SayHello)
-      sayHello(request: HelloRequest): HelloReply {
-        return {message: `hello ${request.name}`};
+    class GreeterCtrl implements Greeter {
+      // Tell LoopBack that this is a Service RPC implementation
+      @grpc(protoMetadata.fileDescriptor)
+      async unaryTest(request: TestRequest): Promise<TestResponse> {
+        return {
+          message: 'Hello ' + request.name,
+        };
       }
 
-      @grpc(Greeter.SayTest)
-      sayTest(request: TestRequest): TestReply {
+      @grpc(protoMetadata.fileDescriptor)
+      async clientStreamTest(
+        request: Observable<TestRequest>,
+      ): Promise<TestResponse> {
+        const names: string[] = [];
+        let error;
+        const observer: PartialObserver<TestRequest> = {
+          next: (value) => names.push(value.name),
+          error: (err) => (error = err),
+        };
+        request.subscribe(observer);
+        await lastValueFrom(request);
+        if (error) {
+          throw error;
+        }
         return {
-          message: 'Test ' + request.name,
+          message: names.join(' '),
         };
```

Comment:
I don't have an in-depth knowledge on rxjs; Is there a benefit over the previous "plain functions" for gRPC?

Reply:
I don't think it adds anything more than encapsulation. I have updated my PR, using implementations of "plain functions" for `@gRPC/grpc-js`